File: NodeApp/service.mjs
import { nanoid } from "nanoid";
import { save } from "./db.mjs";
const baseUrl = "http://test.short/";

export async function shorten(srcUrl) {
  if (!srcUrl) {
    return;
  }

  const urlId = nanoid(10);
  const shortenedUrl = `${baseUrl}${urlId}`;

  const dbStatus = await save(urlId, srcUrl);
  return dbStatus ? shortenedUrl : undefined;
}

File: GoApp/service.go
package main

import gonanoid "github.com/matoous/go-nanoid/v2"

var baseUrl = "http://test.short/"

func Shorten(srcUrl string) string {

	id, _ := gonanoid.New(10)
	shortenedUrl := baseUrl + id
	Save(id, srcUrl)
	return shortenedUrl
}
